api/handlers.go

```diff
@@ -489,6 +489,34 @@ func postCommit(c *context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// POST /containers/{name:.*}/rename
+func postRename(c *context, w http.ResponseWriter, r *http.Request) {
+	container, err := getContainerFromVars(c, mux.Vars(r))
+	if err != nil {
+		httpError(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	if err := r.ParseForm(); err != nil {
+		httpError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// check new name whether avaliable
+	newName := r.Form.Get("name")
+	if conflictContainer := c.cluster.Container(newName); conflictContainer != nil {
```

Comment:
I define a Conflict Error when check name in cluster. Where to put better? In cluster/cluster.go Or in cluster/swarm/cluster.go?
Also , i see the flow of create container. it also check name in api/handler.go. Need to fix?